Solved locationTiers persistency issue

shivam-1810 opened the pull request:
As required, locationTiers has been made persistent, now, it shouldn't change after app reloads and restarts. Let me know if more changes are required, thanks!

jacobthejones commented on Flutter/shelter_partner/lib/repositories/auth_repository.dart:
```diff
@@ -158,6 +158,7 @@ class AuthRepository {
       appendAnimalDataToURL: false,
       removeAds: true,
       simplisticMode: true,
+      locationTierCount: 2,
```
I'd think 4 to match the current default
```suggestion
      locationTierCount: 4,
```

jacobthejones commented on Flutter/shelter_partner/lib/views/pages/enrichment_page.dart:
```diff
@@ -426,6 +426,8 @@ class _EnrichmentPageState extends ConsumerState<EnrichmentPage>
     final shelterSettings = ref.watch(shelterSettingsViewModelProvider);
     final accountSettings = ref.watch(accountSettingsViewModelProvider);
+    selectedLocationTierCount =
+        accountSettings.value?.accountSettings?.locationTierCount ?? 2;
```
Current default is 4 so I'd assume stick with that here as well unless you've heard otherwise
```suggestion
        accountSettings.value?.accountSettings?.locationTierCount ?? 4;
```

jacobthejones commented on Flutter/shelter_partner/lib/views/pages/enrichment_page.dart:
```diff
@@ -596,10 +598,21 @@ class _EnrichmentPageState extends ConsumerState<EnrichmentPage>
+                                            .value
+                                            ?.accountSettings
+                                            ?.locationTierCount ??
+                                        2,
```
```suggestion
                                        4,
```

jacobthejones replied:
Because the default is used in many places, it would also make sense to have a variable somewhere so that if we need to change the default in the future, we only need to change it in one place.

shivam-1810 commented:
@jacobthejones I have changed the default to 4 everywhere. Also, the one variable selectedLocationTierCount in the enrichment_page.dart can be used to change the default value if required without thinking of other variables in the backend, that's because that's the only variable which manages how the page should look, the backend variables have just been initialized with 4.
So, if in future we changed default to 2 or 3, it would reflect on the page and also won't be affected by the backend variable initiation with 4, however even modifying that for a single time will modify the variable in backend too. So, I guess this can be fine, suggest me another way, if it doesn't seem good.

jacobthejones commented:
New changes look good to me! I think you'll still need an approval from Jared, since he's still the repo owner and I don't currently have write access.

shivam-1810 commented:
Sure, thanks for the suggested ideas and changes though!
